### PR TITLE
Client Integration

### PR DESCRIPTION
**Pull request by pzalep1**
This PR puts the finishing touches on the service in order for it to integrate with our client.

### CODE REVIEW

File: src/outcomes/outcomes.controller.ts [hunk 1]
```diff
@@ -177,17 +205,18 @@ export class OutcomesController {
     };
   }
 
-  async getLearningObject(username: string, learningObjectID: string) {
+  async getLearningObject(username: string, learningObjectID: string, user: any) {
     try {
       const response = await request
-        .get(`${process.env.LEARNING_OBJECT_SERVICE_API}/users/${username}/learning-objects/${learningObjectID}/outcomes`)
```
Comment (Olaniyan99) [inline src/outcomes/outcomes.controller.ts:183]: why is /outcomes removed, a lil confused. is there a reason for the change?

File: src/outcomes/outcomes.controller.ts [hunk 2]
```diff
@@ -206,16 +235,12 @@ export class OutcomesController {
 
   async getGuideline(guidelineID: string) {
     const guideline = await this.outcomeService.getGuideline(guidelineID);
-
-    if (!guideline) {
-      throw new NotFoundException('The specified Guideline was not found');
-    }
-
+
```
Comment (Olaniyan99) [inline src/outcomes/outcomes.controller.ts:238]: what happens if guideline isnt found?

File: src/outcomes/outcomes.controller.ts [hunk 3]
```diff
@@ -177,17 +205,18 @@ export class OutcomesController {
...
       const response = await request
-        .get(`${process.env.LEARNING_OBJECT_SERVICE_API}/users/${username}/learning-objects/${learningObjectID}/outcomes`)
+        .get(`${process.env.LEARNING_OBJECT_SERVICE_API}/users/${username}/learning-objects/${learningObjectID}`)
         .set('Accept', 'application/json')
+        .set('Authorization', user)
 
       return response.body;
 
     } catch (error) {
       if (error.status === 401) {
```
Comment (Olaniyan99) [inline src/outcomes/outcomes.controller.ts:218]: should this be throwing a 403? the way i understand is a 403 response is the server telling you, “I’m sorry. I know who you are–I believe who you say you are–but you just don’t have permission to access this resource. i might be wrong tho. 401 meaning the user is strictly unauthorized. what do u think?

**Review by Olaniyan99 — APPROVED** [review]

**Review by davidnso — APPROVED** [review]
This looks so much like spring.

**Comment by pzalep1** [comment]
> This looks so much like spring.

But its not Java.

**Comment by Olaniyan99** [comment]
yea i agree, nestjs is very similar to spring with the use of decorators

**Pull request merged**